src/Models/Session.ts

```diff
@@ -0,0 +1,72 @@
+import {TelemetryMessage} from "./TelemetryMessage";
+
+export enum Platform {
+    Linux = "Linux",
+    MacOS = "macOS",
+    Unknown = "Unknown"
+}
+
+export interface PlatformInfo {
+    variant?: string;
+    distro?: string;
+    version?: string;
+}
+
+export class Session {
+    id: string;
+    userId: string;
+    version: string;
+    startTime: number;
+    endTime?: number;
+    duration?: number;
+    backendPlatform: Platform;
+    backendPlatformInfo: PlatformInfo;
+    // TODO: Add frontend info
+    frontendPlatform?: string;
+    frontendUserAgent?: string;
+
+    constructor(entry: TelemetryMessage, userId: string) {
+        this.id = entry.sessionId;
+        this.userId = userId;
+        this.version = entry.version;
+        this.startTime = entry.timestamp;
+        const platform = entry.details?.platformInfo?.platform;
+        const platformInfoString = entry.details?.platformInfo?.release_info;
+
+        if (platformInfoString) {
+            this.backendPlatformInfo = Session.GetPlatformInfo(platformInfoString, platform);
+        } else {
+            this.backendPlatformInfo = {};
+        }
+
+        if (platform) {
+            this.backendPlatform = platform;
+        } else {
+            this.backendPlatform = Platform.Unknown;
+        }
+    }
+
+    private static GetPlatformInfo(infoString: string, platform: Platform): PlatformInfo {
+        if (!infoString || !platform) {
+            return {};
+        }
+
+        if (platform === Platform.Linux) {
+            // Pull out entries from /etc/os-release string, and trim quotation marks from any elemets
+            const nameRegex = /NAME=(?<name>.+?)\n/m;
+            const variantRegex = /ID_LIKE=(?<variant>.+?)\n/m;
+            const versionRegex = /VERSION_ID=(?<version>.+?)\n/m;
```

Comment:
Why not add `"?` around the patterns in these to avoid stripping off quotation marks in a separate step? This works for me when I test it with a non-greedy pattern and an explicit end anchor, both of which you already have (although Python `re` syntax may behave slightly differently).

I wouldn't use named groups in regexes with only one group each, for simplicity. I would probably also iterate over the file once to read the data into a key/value structure line by line, and then access the parsed values from the structure. But this is bikeshedding; if you remove the additional quote regex, the verbosity will probably be the same either way. :)